### PR TITLE
[#438] Initialize autocomplete terms

### PR DESCRIPTION
**Pull request by fenekku**
*This is the first of 2 PRs across 2 git repositories to address the problems we have been having with the Subjects (terms)*

When approved, instead of merging this PR, I will completely override the `our_master` branch by it, because this PR starts from the version of the library that we used (and know works), v0.0.6.

PR content:
- **Main**: addtion of autocompleteTerms that deals with initializing autocomplete terms properly and with less network requests.
- **Other**: Add `dist/` content because it is needed for an NPM package installed from GitHub
- **Other**: Update dependencies because tests/library couldn't run

### CODE REVIEW

File: karma-src.conf.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ module.exports = function(config) {
       'karma-jquery',
       'karma-mocha',
       'karma-ng-html2js-preprocessor',
-      'karma-phantomjs-launcher',
+      'karma-chrome-launcher',
```
Comment (sharpattack) [inline karma-src.conf.js:49]: This is for testing, right? Why this change?

File: src/invenio-records-js/directives/invenioRecordsForms.js [hunk 2]
```diff
@@ -190,10 +190,42 @@ function invenioRecordsForm($q, schemaFormDecorators, InvenioRecordsAPI,
...
+            'value': model
+          }
+        ];
+        var defer = $q.defer();
+        defer.resolve({'data': titleMap});
+        return defer.promise;
+      } else {
+        return autocompleteSuggest(options, query);
+      }
+    }
```
Comment (sharpattack) [inline src/invenio-records-js/directives/invenioRecordsForms.js:223]: Looking forward to the next PR to get a better grasp on this.

**Review by sharpattack — APPROVED** [review]
Looks good, just curious why the change from PhantomJS to Chrome. 

About overriding the branches, why not make the PR to `our_master` branch with the necessary changes? I am fine with this, but I figure it would be easier to maintain and keep the history.

**Comment by fenekku** [comment]
> why not make the PR to our_master branch with the necessary changes?

Let me unpack the

> because this PR starts from the version of the library that we used (and know works), v0.0.6.

We have been using `invenio-records-js@0.0.6` since the beginning of the project. That version has its quirks, but it works with our codebase; we have explicitly coded for its data exchange interface. When making the previous changes, I started from the repo's master branch which is at `0.0.8`, thinking nothing of it. Unfortunately the data exchange interface changed between versions. It would require more effort / risk than what we want given the coming Fall meeting to change our codebase now (AND, as usual, this library is going away this year). I though it better to start from a known working version and add our needed functionality + fix libraries/tests.

**Comment by fenekku** [comment]
> why the change from PhantomJS to Chrome

PhantomJS is not supported anymore: https://github.com/ariya/phantomjs/issues/15344
Chrome headless is fast and does what we want (could have also used firefox headless I guess).

File: karma-src.conf.js (hunk 1 above)
Comment (fenekku) [reply karma-src.conf.js:49]: Yes it is for tests. PhantomJS is not supported anymore: ariya/phantomjs#15344. Running it breaks a bunch of stuff.

File: src/invenio-records-js/directives/invenioRecordsForms.js (hunk 2 above)
Comment (fenekku) [reply src/invenio-records-js/directives/invenioRecordsForms.js:223]: Yes, it will be clearer I hope.
